api/db-cdr/changelog-schema/db.changelog-9-concept-count-cols.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd
+        http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+  <changeSet author="biopete" id="changelog-9-concept-count-columns added">
+
+    <addColumn tableName="concept" schemaName="cdr">
```

Comment:
Can this column be included in the create of the table, instead of modifying it?

Reply:
I thought that is not how we do liquibase. Never modify a changelog file. Make a new one. Stuff breaks otherwise. The old files don't get rerun in schema .

Reply:
The reason to have many change sets in liquibase is primarily to migrate a schema over time, when you have multiple commits to master, and need to update existing databases from previous commits.

When you have one big commit like this, it's probably better to combine everything into a single change set, or at a minimum all the changes for each table should be in the same change set, to make it easier to understand.

In some cases, we have changesets where runAlways="true":

https://github.com/all-of-us/workbench/blob/master/api/db-cdr/changelog-data/db.changelog-1.xml#L8

Those will run even when databases have already run that change set, and try to update tables to match the schema. (Though generally it's better not to do this I think.)

FWIW: prior to launch, we're likely to reset our databases and combine change sets together, so we can have a schema in liquibase that's easier to understand. Splitting things up over a bunch of change sets is hard to grok.

Reply:
I will optimize the changelogs when we finalize db like we discussed dan.